src/NuGet.Core/NuGet.DependencyResolver.Core/GraphModel/GraphOperations.cs

```diff
@@ -774,6 +817,75 @@ private static CyclesAndDowngrades CreateState(List<GraphNode<RemoteResolveResul
             };
         }
 
+        private static void DetectAndMarkAmbiguousCentralTransitiveDependencies<TItem>(Tracker<TItem> tracker, List<GraphNode<TItem>> centralTransitiveNodes)
+        {
+            // if a central transitive node has all parents disputed or ambiguous mark it and its children ambiguous
+            int ctdCount = centralTransitiveNodes.Count;
+            for (int i = 0; i < ctdCount; i++)
+            {
+                if (centralTransitiveNodes[i].Disposition == Disposition.Acceptable)
+                {
+                    bool allParentsAreDisputedOrAmbiguous = !centralTransitiveNodes[i].ParentNodes
+                        .Any(p => p.Disposition != Disposition.Rejected && !(tracker.IsDisputed(p.Item) || tracker.IsAmbiguous(p.Item)));
+
+                    if (allParentsAreDisputedOrAmbiguous)
+                    {
+                        // children of ambiguous nodes were already marked as ambiguous, skip them
+                        centralTransitiveNodes[i].ForEach(x => tracker.MarkAmbiguous(x.Item), pn => tracker.IsAmbiguous(pn.Item));
+                    }
+                }
+            };
+        }
+
+        private static void RejectCentralTransitiveBecauseOfRejectedParents<TItem>(this GraphNode<TItem> root, Tracker<TItem> tracker, List<GraphNode<TItem>> centralTransitiveNodes)
+        {
+            HashSet<GraphNode<TItem>> internalContext = new HashSet<GraphNode<TItem>>();
+
+            // reject nodes of rejected nodes for the central transitive nodes and track nodes that were not yet rejected
+            // as more nodes can be rejected do not track the nodes until all the rejects are completed
+            int ctdCount = centralTransitiveNodes.Count;
+            for (int i = 0; i < ctdCount; i++)
+            {
+                centralTransitiveNodes[i].ForEach(root.Disposition != Disposition.Rejected, (node, state, context) => WalkTreeRejectNodesOfRejectedNodes(state, node, context), internalContext);
+            }
+
+            // If a node has its parents rejected, reject the node and its children
+            // Need to do this in a loop because more nodes can be rejected as their parents become rejected
+            bool rejectsPerformed = true;
```

Comment:
naming: This is about whether there's any pending rejections. Given that we reset it when there's  nothing left, feels like it'd be easier to read if named as such.

Reply:
Updated